Adding version check for hypothesis deadline

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31262 Adding version check for hypothesis deadline**

Differential Revision: [D19036700](https://our.internmc.facebook.com/intern/diff/D19036700)

Commit:
Adding version check for hypothesis deadline

jamesr66a commented on test/hypothesis_utils.py:
```diff
@@ -304,11 +305,18 @@ def tensor_conv(
+else:
+    from warnining import warn
+    warn("Your hypothesis version is outdated and doesn't support `deadline`. "
+         "Some tests are likely to fail with `DeadlineExceeded` error. "
```
This seems like a contradiction with the sentence before it

z-a-f replied:
Oh it does read like it. I should add that the `deadline` is disabled, and now the tests can fail with...

jamesr66a reviewed (commented):
We should probably just require a recent Hypothesis version rather than adding complexity to our codebase. Is this just needed because of the default `hypothesis` version on devservers?

z-a-f commented:
> We should probably just require a recent Hypothesis version rather than adding complexity to our codebase. Is this just needed because of the default `hypothesis` version on devservers?

Yes -- I cannot run most tests on devserver because of the versions

Commit:
Update on "Adding version check for hypothesis deadline"


Differential Revision: [D19036700](https://our.internmc.facebook.com/intern/diff/D19036700)

Commit:
Update on "Adding version check for hypothesis deadline"


Differential Revision: [D19036700](https://our.internmc.facebook.com/intern/diff/D19036700)

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit ad1d80f1:


* **1/1** broken upstream at merge base 68e51723 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=68e517238291d1149a89d1f4e9fc90b560804d00&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=ad1d80f1e5a170785162ac590aea8f674728b25d).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_clang5_asan_test](https://dr.pytorch.org/build-details.html?build_id=78587548)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 commented on test/hypothesis_utils.py:
```diff
@@ -304,11 +305,18 @@ def tensor_conv(
+# Creating (and loading) a separate profile overrides any settings the user
+# already specified.
+settings._profiles[settings._current_profile].__dict__['deadline'] = None
+settings._profiles[settings._current_profile].__dict__['timeout'] = hypothesis.unlimited
```
is `hypothesis.unlimited` valid for hypothesis version < 3.27.0?

z-a-f replied:
Good point!

Commit:
Update on "Adding version check for hypothesis deadline"


Differential Revision: [D19036700](https://our.internmc.facebook.com/intern/diff/D19036700)

jerryzh168 reviewed (approved):
LGTM
